public/static-checkout/js/scripts.js

```diff
@@ -0,0 +1,313 @@
+  }
+}
+
+function renderCountryState() {
+  countrySelect.onchange = function getCountryStates() {
+    var stateSelectHtml = document.createElement('div');
+    stateSelectHtml.innerHTML = '<select id="state" name="state"><option value="" selected="selected">Choose state/province</option></select>';
+    stateSelect.length = 1;
+    if (!(countrySelect.value === 'CA' || countrySelect.value === 'US')) {
+      stateSelectHtml = document.createElement('div');
+      stateSelectHtml.innerHTML = '<input id="state" type="text" name="state">';
+      stateSelect.parentNode.replaceChild(stateSelectHtml, stateSelect);
+      stateSelect = document.getElementById('state');
+    } else {
+      stateSelect.parentNode.replaceChild(stateSelectHtml, stateSelect);
+      stateSelect = document.getElementById('state');
+      if (countrySelect.value === 'CA') {
+        renderStates('CA');
+      } else if (countrySelect.value === 'US') {
+        renderStates('US');
+      }
+    }
+  };
+}
+
+document.addEventListener('DOMContentLoaded', function () {
+  renderCountryState();
+});
+
+couponButton.addEventListener('click', function couponRedirectToMicrofrontend(event) {
+  event.preventDefault();
+  var couponCode = document.getElementById('couponcode').value;
+  redirectToMFE(couponCode);
+}, false);
+
+
+function parseCookies() {
+  var cookies = {};
+  if (document.cookie && document.cookie !== '') {
+    var cookiesArray = document.cookie.split('; ');
+    for (var i = 0; i < cookiesArray.length; i++) {
+      var m = cookiesArray[i].trim().match(/(\w+)=(.*)/);
+      if (m !== undefined) {
+        cookies[m[1]] = decodeURIComponent(m[2]);
+      }
+    }
+  }
+  return cookies;
+}
+
+function disableForm() {
+  var elementsToDisable = document.querySelectorAll('input, select, button');
+  for (var i = 0; i < elementsToDisable.length; i++) {
+    elementsToDisable[i].disabled = true;
+  }
+}
+
+paypalButton.addEventListener('click', function submitPaypal() {
+  disableForm();
+  var cookies = parseCookies();
+  var xhr = new XMLHttpRequest();
+  xhr.open('POST', ecommerceBaseUrl + '/api/v2/checkout/', true);
+  xhr.setRequestHeader('Content-Type', 'application/json');
+  xhr.setRequestHeader('X-CSRFToken', cookies.csrftoken);
+  xhr.setRequestHeader('USE-JWT-COOKIE', true);
+  xhr.withCredentials = true;
+  // eslint-disable-next-line func-names
+  xhr.onreadystatechange = function () {
+    if (this.readyState === XMLHttpRequest.DONE) {
+      if (this.status === 200) {
+        var form = document.createElement('form');
+        form.method = 'POST';
+        form.action = JSON.parse(this.response).payment_page_url;
+        document.body.appendChild(form);
+        form.submit();
+      } else {
+        redirectToMFE();
+      }
+    }
+  };
+  var basketId = getBasketId();
+  xhr.send(JSON.stringify({
+    basketId,
+    payment_processor: 'paypal',
+  }));
+});
+
+checkoutForm.addEventListener('submit', function submitCybersource(event) {
```

Comment:
is this just copying what the MFE is doing? this code seems rather complex so i would have more questions if it is not

Reply:
Emma can better answer this, but we can discuss on slack

Reply:
It is. :)